frontend/lib/src/components/widgets/CustomComponent/ComponentInstance.tsx

```diff
@@ -235,6 +235,30 @@ function ComponentInstance(props: Props): ReactElement {
     })
   }, COMPONENT_READY_WARNING_TIME_MS)
 
+  useEffect(() => {
+    // Iframe has no onerror event - so check custom component
+    // src on mount to catch iframe load errors
+    registry.checkSourceResponse(
+      getSrc(componentName, registry, url),
+      componentName
+    )
+    // Array must be empty to run as mount/cleanup
+    // TODO: Update to match React best practices
+    // eslint-disable-next-line react-compiler/react-compiler
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
What if the url changes like if I did:

```python
if some_check:
  custom_component_1()
else:
  custom_component_2()
```

I imagine with this code, the component remains mounted, but the second custom component will not be checked.

Should we just check the src url and check if it changes.

```javascript
const componentSourceUrl = getSrc(componentName, registry, url)

useEffect(() => {
  registry.checkSourceResponse(componentSourceUrl, componentName)
}, [componentSourceUrl, componentName])
```